pkg/controller/perconaservermongodb/ssl.go

```diff
@@ -20,19 +20,46 @@ import (
 
 func (r *ReconcilePerconaServerMongoDB) reconsileSSL(cr *api.PerconaServerMongoDB) error {
 	secretObj := corev1.Secret{}
-	err := r.client.Get(context.TODO(),
+	secretInternalObj := corev1.Secret{}
+	errSecret := r.client.Get(context.TODO(),
 		types.NamespacedName{
 			Namespace: cr.Namespace,
 			Name:      cr.Spec.Secrets.SSL,
 		},
 		&secretObj,
 	)
-	if err == nil {
+	errInternalSecret := r.client.Get(context.TODO(),
+		types.NamespacedName{
+			Namespace: cr.Namespace,
+			Name:      cr.Spec.Secrets.SSL + "-ssl-internal",
+		},
+		&secretInternalObj,
+	)
+	if errSecret == nil && errInternalSecret == nil {
 		return nil
-	} else if !k8serrors.IsNotFound(err) {
-		return fmt.Errorf("get secret: %v", err)
+	} else if errSecret != nil && !k8serrors.IsNotFound(errSecret) {
+		return fmt.Errorf("get secret: %v", errSecret)
+	} else if errInternalSecret != nil && !k8serrors.IsNotFound(errInternalSecret) {
+		return fmt.Errorf("get internal secret: %v", errInternalSecret)
+	}
+	// don't create secret ssl-internal if secret ssl is not created by operator
+	if errSecret == nil {
+		ownedByCr := false
+		ownerCr, err := OwnerRef(cr, r.scheme)
+		if err != nil {
+			return err
+		}
+		for _, owner := range secretObj.OwnerReferences {
+			if owner.UID == ownerCr.UID {
+				ownedByCr = true
+				break
+			}
+		}
+		if !ownedByCr {
+			return nil
+		}
```

Comment:
it's better to use https://pkg.go.dev/k8s.io/apimachinery/pkg/apis/meta/v1#IsControlledBy

Reply:
Fixed